src/main/java/de/hysky/skyblocker/utils/Utils.java

```diff
@@ -419,12 +393,22 @@ case HelloS2CPacket(var environment) -> {
 
             case LocationUpdateS2CPacket(var serverName, var serverType, var _lobbyName, var mode, var map) -> {
                 Utils.server = serverName;
+                String previousServerType = Utils.gameType;
                 Utils.gameType = serverType.orElse("");
                 Utils.locationRaw = mode.orElse("");
                 Utils.location = Location.from(locationRaw);
                 Utils.map = map.orElse("");
 
                 SkyblockEvents.LOCATION_CHANGE.invoker().onSkyblockLocationChange(location);
+
+                if (Utils.gameType.equals("SKYBLOCK")) {
+                    isOnSkyblock = true;
+
+                    if (!previousServerType.equals("SKYBLOCK")) SkyblockEvents.JOIN.invoker().onSkyblockJoin();
+                } else {
+                    isOnSkyblock = false;
+                    SkyblockEvents.LEAVE.invoker().onSkyblockLeave();
```

Comment:
am i dumb or will this fire every time you go from a non skyblock server to another non skyblock server? wouldn't it need a
```java
if (previousServerType.equals("SKYBLOCK"))
```
or something similar?

Reply:
no? the `else` is for the first `if`